src/display/editor/ink.js

```diff
@@ -0,0 +1,689 @@
+    ctx.setTransform(1, 0, 0, 1, 0, 0);
+    ctx.clearRect(0, 0, width, height);
+    this.#updateTransform();
+    for (const path of this.bezierPath2D) {
+      ctx.stroke(path);
+    }
+  }
+
+  /**
+   * Commit the curves we have in this editor.
+   * @returns {undefined}
+   */
+  commit() {
+    if (this.#disableEditing) {
+      return;
+    }
+    this.disableEditMode();
+
+    this.#disableEditing = true;
+    this.div.classList.add("disabled");
+
+    this.#fitToContent();
+  }
+
+  /** @inheritdoc */
+  focusin(/* event */) {
+    super.focusin();
+    this.enableEditMode();
+  }
+
+  /**
+   * onmousedown callback for the canvas we're drawing on.
+   * @param {MouseEvent} event
+   * @returns {undefined}
+   */
+  canvasMousedown(event) {
+    if (!this.isInEditMode() || this.#disableEditing) {
+      return;
+    }
+
+    event.stopPropagation();
+
+    this.canvas.addEventListener("mouseleave", this.#boundCanvasMouseleave);
+    this.canvas.addEventListener("mousemove", this.#boundCanvasMousemove);
+
+    this.#startDrawing(event.offsetX, event.offsetY);
+  }
+
+  /**
+   * onmousemove callback for the canvas we're drawing on.
+   * @param {MouseEvent} event
+   * @returns {undefined}
+   */
+  canvasMousemove(event) {
+    event.stopPropagation();
+    this.#draw(event.offsetX, event.offsetY);
+  }
+
+  /**
+   * onmouseup callback for the canvas we're drawing on.
+   * @param {MouseEvent} event
+   * @returns {undefined}
+   */
+  canvasMouseup(event) {
+    if (this.isInEditMode() && this.currentPath.length !== 0) {
+      event.stopPropagation();
+      this.#endDrawing(event);
+    }
+  }
+
+  /**
+   * onmouseleave callback for the canvas we're drawing on.
+   * @param {MouseEvent} event
+   * @returns {undefined}
+   */
+  canvasMouseleave(event) {
+    this.#endDrawing(event);
+  }
+
+  /**
+   * End the drawing.
+   * @param {MouseEvent} event
+   */
+  #endDrawing(event) {
+    this.#stopDrawing(event.offsetX, event.offsetY);
+
+    this.canvas.removeEventListener("mouseleave", this.#boundCanvasMouseleave);
+    this.canvas.removeEventListener("mousemove", this.#boundCanvasMousemove);
+  }
+
+  /** @inheritdoc */
+  render() {
+    if (this.div) {
+      return this.div;
+    }
+
+    super.render();
+    this.canvas = document.createElement("canvas");
+    this.canvas.className = "inkEditorCanvas";
+    this.div.appendChild(this.canvas);
+    this.ctx = this.canvas.getContext("2d");
+
+    this.div.classList.add("editing");
+
+    if (this.width) {
+      // This editor has been created in using copy (ctrl+c).
+      this.setAt(this.x + this.width, this.y + this.height);
+      this.setDims(this.width, this.height);
+      this.#setCanvasDims();
+      this.#redraw();
+    }
+
+    const observer = new ResizeObserver(entries => {
+      const rect = entries[0].contentRect;
+      if (rect.width && rect.height) {
+        this.setDimensions(rect.width, rect.height);
+      }
+    });
+    observer.observe(this.div);
```

Comment:
Is the `observer` removed automatically when the current `InkEditor`-instance is removed, or do we (somehow) need to do that manually?